### PR TITLE
Let MovieFileWriter save temp files in a new dir

### PR DESCRIPTION
**Pull request by fredrik-1**
Another try to get the animation module to work better. The reason for these changes are that they make it possible to create videos of animations with a low frame rate to work in video players like the VLC player. This can be done by changing the output frame rate to a higher value by for example using `extra_args=['-r', '25']` or `extra_args=['-vf', 'fps=25']`.

The changes in this pr:
- deleted `-vframes` in ffmpeg file writer because it is usually unnecessary and make it impossible to change the output frame rate. 
- made the `MovieFileWriter` save the temporary files in a new directory to avoid the possible problem that the writer reads old files.
- made it possible to not have anything except `extra_args` after the `-i temp_files` by not using a codec if `codec=''`. This is not really necessary but gives more freedom and might be useful for someone.

### CODE REVIEW

File: lib/matplotlib/animation.py [hunk 1]
```diff
@@ -542,6 +549,9 @@ def cleanup(self):
                        self._temp_names)
             for fname in self._temp_names:
                 os.remove(fname)
+            folder = os.path.split(fname)[0]
+            if folder is not '':
+                shutil.rmtree(folder, ignore_errors=True)
```
[the remark this answers is not in the document]
Comment (fredrik-1) [reply lib/matplotlib/animation.py]: Good point. That code is definitely dangerous. I have changed to `os.rmdir` now.

One problem is that I believe that `os.rmdir` sometimes don't remove empty directory's on windows for some reason. It seems to work for me now and it is possible that it doesn't happen that often and never in this application.

**Comment by fredrik-1** [comment]
Made the suggested changes.

**Comment by anntzer** [comment]
See also https://github.com/matplotlib/matplotlib/pull/3536 for what was one of my first PRs to matplotlib :p
I think nowadays I'd suggest using TemporaryDirectory instead, it's more robust wrt. interpreter shutdown.

**Comment by jklymak** [comment]
@fredrik-1 did you want to proceed w/ this?  We are trying to tag 3.1 soonish...

**Comment by jklymak** [comment]
Thanks a lot for the contribution.  I'm going to close this as having had no action for quite a while.  OTOH, if there is still a community that wants this, please feel free to re-open, with my apologies!

**Pull request closed without merging**